examples/PSyclone/psyclone.py

```diff
@@ -100,32 +108,37 @@ def _recurse(self, siblings, my_index, options, all_options, invoke):
         # siblings includes this loop
         n_siblings = len(siblings)
         index = my_index+1
-        while index<n_siblings:
+        while index < n_siblings:
             try:
                 trans._validate(siblings[index-1], siblings[index])
                 my_options.append([siblings[index-1], siblings[index]])
                 #print "Fusion for {0} possible".format(my_options)
                 if my_options:
-                    all_options.append({invoke:list(my_options)})
-                self._recurse(siblings, index+1, my_options, all_options, invoke)
+                    all_options.append({invoke: list(my_options)})
+                self._recurse(siblings, index+1, my_options, all_options,
+                              invoke)
             except TransformationError:
                 break
             index += 1
 
     def __init__(self, dependent_invokes=False):
-        self.state = True # tell melody that I expect state data (psy) to be passed
+        # tell melody that I expect state data (psy) to be passed
+        self.state = True
         self._name = "Loop Fusion"
         self._dependent_invokes = dependent_invokes
 
-    def options(self, psy):
-        ''' '''
-        # compute options dynamically here as they may depend on previous changes to the psy tree
+    def options(self, my_psy):
+        '''Returns all potential loop fusion options for the psy object
+        provided'''
+        # compute options dynamically here as they may depend on previous
+        # changes to the psy tree
         my_options = []
-        invokes = psy.invokes.invoke_list
+        invokes = my_psy.invokes.invoke_list
         #print "there are {0} invokes".format(len(invokes))
         if self._dependent_invokes:
-            print ("dependent invokes assumes fusion in one invoke might affect fusion "
-                   "in another invoke. This is not yet implemented")
+            print ("dependent invokes assumes fusion in one invoke might "
+                   "affect fusion in another invoke. This is not yet "
+                   "implemented")
             exit(1)
```

Comment:
Again, this code does not currently work so I've not spent much time on its niceties. However, I've made it raise a RuntimeError exception.